HelpSource/Reference/AudioDeviceSelection.schelp

```diff
@@ -0,0 +1,172 @@
+code::
+Server.default.options.inDevice_("Built-in Microph");
+Server.default.options.outDevice_("Built-in Output");
+::
+
+In case of a dedicated audio interfaces, you might need to specify a single device, e.g.
+code::
+Server.default.options.device_("MOTU 828");
+::
+
+On macOS you can programmatically obtain list of available audio devices without booting the server:
+code::
+ServerOptions.devices; //all devices
+ServerOptions.inDevices; //input devices
+ServerOptions.outDevices; //output devices
+::
+
+subsection:: Sample rate mismatch
+
+One of the reasons for the server's failure to boot is the mismatch between input and output devices' sample rate. This is a possibility when using a different device for input and output (which is the case when using the internal soundcard). If the server fails to boot due to sample rate mismatch, an error will be posted accordingly.
+
+You should set both input and output devices' sample rate to the same value. You can do this in Audio MIDI Setup:
+code::
+"open -a 'Audio MIDI Setup'".unixCmd; // execute this to launch it
+::
+
+subsection:: Aggregate device
+
+Sometimes you might want to use multiple devices for input or output. In such case, macOS provides a functionality to combine multiple physical devices into one called an teletype::Aggregate Device::. To create one, you have to open the Audio MIDI Setup application (in /Applications/Utilities). You should do this from an user account with administrator privileges.
+code::
+"open -a 'Audio MIDI Setup'".unixCmd; // execute this to launch it
+::
+LIST::
+##In the teletype::Audio Devices:: window click on the Plus button on the bottom left and choose teletype::Create Aggregate Device::.
+##You can change its name to something specific if desired, for example "InternalAndMOTU".
+##Check the Use column on the right for the devices you want to combine.
+::
+Now you need to tell SuperCollider to use your new aggregate device.
+code::
+Server.default.options.device = "Aggregate Device"; //or the name you have specified in the previous step
+::
+
+After (re)booting the server (code::Server.default.reboot::) you should see in the post window that it now uses the Aggregate Device instead of system defaults:
+teletype::
+[...]
+"Aggregate Device" Input Device
+[...]
+"Aggregate Device" Output Device
+::
+
+Note that when you specify a sound device through SuperCollider's code::ServerOptions::, there is no need to use the aggregate device as the system's default device.
+
+
+section:: Linux
+
+By default SuperCollider on Linux uses JACK and the audio device selection is managed by the JACK server.
+
+subsection:: Specifying JACK server and client name
+
+When the server is compiled for JACK natively, the code::ServerOption::'s teletype::device:: can be used to connect to a named server and
+use a specific client name. The argument specifies either a server name or a server name and the requested client
```

Comment:
Seems to be some confusion about what is the "server" here.

From JACK's point of view, *JACK* is the server and apps using it (e.g. scsynth) are clients. From our point of view, scsynth is the server.

If the device is a single name, then it specifies the name under which scsynth or supernova will appear *as a JACK client*.

```
s.options.device = "my_synth";
s.boot;

Booting server 'localhost' on address 127.0.0.1:57110.
JackDriver: client name is 'my_synth'
```

It isn't looking for a JACK server named 'my_synth' -- it's registering with the default JACK server, giving scsynth the name 'my_synth'. `The argument specifies either a server name...` suggests the opposite.

I haven't tried using named JACK servers, so I can't comment on that except that `s.options.device = "aNonExistentJACKServer:my_synth"` does ignore the already-running JACK server and it tries to start a new one. So, definitely, `"JACKServerName:scsynthName"` seems to be a valid format.

(I don't blame anyone for mixing this up... it's very confusing that we have two executables, both of which call themselves a "server.")